runhouse/resources/hardware/cluster.py

```diff
@@ -816,14 +816,14 @@ def restart_server(
         """
         logger.info(f"Restarting Runhouse API server on {self.name}.")
 
-        default_env = _get_env_from(self._default_env) if self._default_env else None
-        if default_env:
-            self._sync_default_env_to_cluster()
-
         if resync_rh:
             self._sync_runhouse_to_cluster(_install_url=_rh_install_url)
             logger.debug("Finished syncing Runhouse to cluster.")
 
+        default_env = _get_env_from(self._default_env) if self._default_env else None
+        if default_env:
+            self._sync_default_env_to_cluster()
```

Comment:
`_sync_default_env_to_cluster` assumed runhouse wasn't started yet so is a special case of installing/syncing through ssh rather than rh. wondering if directly doing `default_env.to(cluster)` would directly work (and we could eventually clean up those functions if so)

Reply:
Runhouse still isn't started, just synced